scripts/deploy/09_eth_put_theta_vault.ts

```diff
@@ -0,0 +1,104 @@
+import { run } from "hardhat";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import {
+  CHAINID,
+  WETH_ADDRESS,
+  USDC_ADDRESS,
+} from "../../constants/constants";
+import {
+  AUCTION_DURATION,
+  MANAGEMENT_FEE,
+  PERFORMANCE_FEE,
+  PREMIUM_DISCOUNT,
+} from "../utils/constants";
+
+const TOKEN_NAME = {
+  [CHAINID.ETH_MAINNET]: 'Ribbon USDC Theta Vault ETH Put',
+  [CHAINID.ETH_KOVAN]: 'Ribbon USDC Theta Vault ETH Put',
+  [CHAINID.AVAX_MAINNET]: 'Ribbon USDC Theta Vault AVAX Put',
+  [CHAINID.AVAX_FUJI]: 'Ribbon USDC Theta Vault AVAX Put',
+};
+
+const TOKEN_SYMBOL = {
+  [CHAINID.ETH_MAINNET]: 'rUSDC-ETH-P-THETA',
+  [CHAINID.ETH_KOVAN]: 'rUSDC-ETH-P-THETA',
+  [CHAINID.AVAX_MAINNET]: 'rUSDC-AVAX-P-THETA',
+  [CHAINID.AVAX_FUJI]: 'rUSDC-AVAX-P-THETA',
+};
+
+const main = async ({
+  network,
+  deployments,
+  ethers,
+  getNamedAccounts,
+}: HardhatRuntimeEnvironment) => {
+  const { BigNumber } = ethers;
+  const { parseEther } = ethers.utils;
+  const { deploy } = deployments;
+  const { deployer, owner, keeper, admin, feeRecipient } =
+    await getNamedAccounts();
+  console.log(`09 - Deploying ETH Put Theta Vault on ${network.name}`);
+
+  const chainId = network.config.chainId;
+
+
+  // Assumes these contracts are already deployed
+  const pricer = await deployments.get("OptionsPremiumPricerETH");
+  const strikeSelection = await deployments.get("StrikeSelectionETH"); //DeltaStrikeSelection
+  const lifecycle = await deployments.get("VaultLifecycle");
+  const logicDeployment = await deployments.get("RibbonThetaVaultLogic");
+  const RibbonThetaVault = await ethers.getContractFactory("RibbonThetaVault", {
+      libraries: { VaultLifecycle: lifecycle.address }
+  });
+
+  const initArgs = [
+    {
+      _owner: owner,
+      _keeper: keeper,
+      _feeRecipient: feeRecipient,
+      _managementFee: MANAGEMENT_FEE,
+      _performanceFee: PERFORMANCE_FEE,
+      _tokenName: TOKEN_NAME[chainId],
+      _tokenSymbol: TOKEN_SYMBOL[chainId],
+      _optionsPremiumPricer: pricer.address,
+      _strikeSelection: strikeSelection.address,
+      _premiumDiscount: PREMIUM_DISCOUNT,
+      _auctionDuration: AUCTION_DURATION,
+      _isUsdcAuction: false,
+      _swapPath: 0x0,
+    },
+    {
+      isPut: true,
+      decimals: 6,
+      asset: USDC_ADDRESS[chainId],
+      underlying: WETH_ADDRESS[chainId],
+      minimumSupply: BigNumber.from(10).pow(10),
+      cap: parseEther("1000"),
```

Comment:
with USDC this cap is 1000 * 10 ** 12 USDC. you should do `BigNumber.from("10000000").mul(BigNumber.from(10).pow(6))` for cap of $10m for example

Reply:
Ah good catch, fixed.